src/java.base/unix/native/libjli/java_md.c

```diff
@@ -498,6 +506,10 @@ GetJREPath(char *path, jint pathsize, jboolean speculative)
     JLI_TraceLauncher("Attempt to get JRE path from launcher executable path\n");
 
     if (GetApplicationHome(path, pathsize)) {
+        if (JLI_IsStaticallyLinked()) {
```

Comment:
`GetJREPath()` does not need to be called for the static case. Any reason why this path is executed for static mode?

Reply:
I guess this is parts of my initial attempt to get the static launcher to work on macOS. I agree that it should not be needed.